sdk/iot/hub/src/az_iot_hub_client_twin.c

```diff
@@ -103,16 +119,119 @@ AZ_NODISCARD az_result az_iot_hub_client_twin_patch_publish_topic_get(
   AZ_PRECONDITION_NOT_NULL(out_mqtt_topic);
   (void)client;
 
-  int32_t required_length = az_span_size(az_iot_hub_twin_patch_pub_topic)
-      + az_span_size(az_iot_hub_client_twin_request_id_suffix) + az_span_size(request_id);
+  int32_t required_length = az_span_size(az_iot_hub_twin_topic_prefix)
+      + az_span_size(az_iot_hub_twin_patch_pub_topic)
+      + (int32_t)sizeof(az_iot_hub_client_twin_question)
+      + az_span_size(az_iot_hub_client_request_id_span)
+      + (int32_t)sizeof(az_iot_hub_client_twin_equals) + az_span_size(request_id);
 
   AZ_RETURN_IF_NOT_ENOUGH_SIZE(mqtt_topic, required_length);
 
-  az_span remainder = az_span_copy(mqtt_topic, az_iot_hub_twin_patch_pub_topic);
-  remainder = az_span_copy(remainder, az_iot_hub_client_twin_request_id_suffix);
-  az_span_copy(remainder, request_id);
+  az_span remainder = az_span_copy(mqtt_topic, az_iot_hub_twin_topic_prefix);
+  remainder = az_span_copy(remainder, az_iot_hub_twin_patch_pub_topic);
+  remainder = az_span_copy_u8(remainder, az_iot_hub_client_twin_question);
+  remainder = az_span_copy(remainder, az_iot_hub_client_request_id_span);
+  remainder = az_span_copy_u8(remainder, az_iot_hub_client_twin_equals);
+  remainder = az_span_copy(remainder, request_id);
 
   *out_mqtt_topic = az_span_slice(mqtt_topic, 0, required_length);
 
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result az_iot_hub_client_twin_received_topic_parse(
```

Comment:
Same comment here for the name of the function, it should change according to C99 discussions, right? `parse` should come before `received`

Reply:
yes agreed. will change